tests/code_update.py

```diff
@@ -43,14 +43,23 @@ def run(args):
 
         LOG.info(f"Adding a node with unsupported code id {new_code_id}")
         try:
-            network.create_and_trust_node(args.patched_file_name, "localhost", args)
+            network.create_and_add_pending_node(
+                args.patched_file_name, "localhost", args, timeout=3
+            )
             assert (
                 False
             ), f"Adding a node with unsupported code id {new_code_id} should fail"
         except TimeoutError as err:
-            assert "CODE_ID_NOT_FOUND" in err.message, err.message
+            joining_errors = getattr(err, "joining_errors", None)
+            if joining_errors is not None:
+                assert "CODE_ID_NOT_FOUND" in joining_errors, joining_errors
+            else:
+                raise
+
+        # Slow quote verification means that any attempt to add a node may cause an election, so confirm primary after adding node
+        primary, others = network.find_primary()
 
-        network.consortium.add_new_code(1, primary, new_code_id)
+        network.consortium.add_new_code(0, primary, new_code_id)
```

Comment:
Why did that go down to 0? What is it?

Reply:
This is the ID of the member that we use to propose the new change. We went from 1-indexed to 0-indexed a while back but haven't updated all of these. I think the actual fix should be to remove the arg in most places (default to proposing as member 0), but its an unnecessary change regardless so I've removed it from this PR.